tests/local/pool.test.ts

```diff
@@ -570,37 +410,132 @@ describe('FXPool', () => {
+    expect(previousFeeBalance).to.be.not.equals(0)
+    console.log('Total unclaimed fees in numeraire before swap: ', formatEther(previousFeeBalance))
 
-  it('can pause pool', async () => {
-    expect(await fxPool.paused()).to.be.equals(false)
+    const fxPHPAmountToSwapInEtherDeposit = 1000
+    const fxPHPDecimals = 18
+    const fxPHPAddress = testEnv.fxPHP.address
+    const usdcAddress = testEnv.USDC.address
 
-    await expect(fxPool.setPaused()).to.emit(fxPool, 'Paused').withArgs(adminAddress)
+    await swaps.buildExecute_SingleSwapGivenIn(
+      fxPHPAddress, // fxPHP = token in
+      usdcAddress, // USDC = token out
+      fxPHPAmountToSwapInEtherDeposit,
+      fxPHPDecimals, // specify token in decimals
+      adminAddress,
+      adminAddress, // the account swapping should get the output tokens
+      fxPool,
+      testEnv,
+      log
+    )
+
+    const currentFeeBalanceAfterSwapBeforeDeposit = await fxPool.totalUnclaimedFeesInNumeraire()
+    expect(currentFeeBalanceAfterSwapBeforeDeposit).is.gt(previousFeeBalance)
+
+    const payload = ethers.utils.defaultAbiCoder.encode(
+      ['uint256', 'address[]'],
+      [parseEther(TEST_DEPOSIT_FEES), sortedAddresses]
+    )
+
+    const depositDetails = await fxPool.viewDeposit(parseEther(TEST_DEPOSIT_FEES))
 
-    const amountIn0 = TEST_DEPOSIT_PAUSEABLE
+    const maxAmountsIn = [ethers.constants.MaxUint256, ethers.constants.MaxUint256]
 
-    let fxPHPAddress = ethers.utils.getAddress(testEnv.fxPHP.address)
+    const joinPoolRequest = {
+      assets: sortedAddresses,
+      maxAmountsIn,
+      userData: payload,
+      fromInternalBalance: false,
+    }
+
+    const lpTokensBeforeDeposit = await fxPool.balanceOf(adminAddress)
+    console.log(`Fee after swap: ${currentFeeBalanceAfterSwapBeforeDeposit}`)
+    console.log(`Lp tokens before deposit: ${lpTokensBeforeDeposit}`)
+
+    await expect(testEnv.vault.joinPool(poolId, adminAddress, adminAddress, joinPoolRequest))
+      .to.emit(fxPool, 'OnJoinPool')
+      .withArgs(poolId, depositDetails[0], [depositDetails[1][0], depositDetails[1][1]])
+      .to.emit(fxPool, 'FeesCollected') // Check for _mintProtocolFees() to be trigerred
+      .withArgs(adminAddress, currentFeeBalanceAfterSwapBeforeDeposit)
+      .to.emit(fxPool, 'Transfer') // Check for minting
+      .withArgs(ethers.constants.AddressZero, adminAddress, currentFeeBalanceAfterSwapBeforeDeposit)
+
+    const currentFeeBalanceAfterDeposit = await fxPool.totalUnclaimedFeesInNumeraire()
+    const lpTokensAfterDeposit = await fxPool.balanceOf(adminAddress)
+
+    expect(currentFeeBalanceAfterDeposit).is.equals(0)
+    expect(lpTokensAfterDeposit).to.equals(
+      lpTokensBeforeDeposit.add(currentFeeBalanceAfterSwapBeforeDeposit).add(depositDetails[0])
+    ) // including LP tokens after deposit
+  })
+
+  it('totalUnclaimedFeesInNumeraire must be minted during onExit', async () => {
+    const previousFeeBalance = await fxPool.totalUnclaimedFeesInNumeraire()
```

Comment:
```suggestion
    const previousFeeBalance = await fxPool.totalUnclaimedFeesInNumeraire()
    expect(previousFeeBalance).is.equals(0)
```
@0xAplki just to be airtight?